src/main/scala/scalaz/netty/Client.scala

```diff
@@ -36,6 +36,7 @@ import _root_.io.netty.handler.codec._
 
 private[netty] final class Client(limit: Int) {
   // this isn't ugly or anything...
+  @volatile
```

Comment:
Is this actually necessary?  I spent some time thinking about it, and I can't convince myself one way or another.  Given that this is a high contention reference, putting `@volatile` here probably has some appreciable performance impact.